corehq/apps/locations/models.py

```diff
@@ -331,6 +331,7 @@ def filter_by_user_input(self, domain, user_input):
                     query = query.filter(
                         Q(name__icontains=part) | Q(site_code__icontains=part)
                     )
+                query = query.filter(domain=domain)
```

Comment:
Is my understanding right that this didn't restrict by domain previously, let alone use domain in this function? Was it relying on other filters added elsewhere to do that?

Reply:
Yes. Usually the `accessible_to_user` filter is also used in conjunction with this one. [`CallCenterOptionsController.get_locations_query`](https://github.com/dimagi/commcare-hq/blob/62febbd8187c27bd5eedfc764d924b59426534e9/corehq/apps/callcenter/views.py#L74-L77) is the exception; I'm not sure how that one is/was being filtered by domain.

Reply:
Looks like this used to filter by domain (which is why it's an arg), but I removed that in https://github.com/dimagi/commcare-hq/pull/26671 - must've been an oversight